packages/docs/actions/DownloadCsv.yaml

```diff
@@ -0,0 +1,104 @@
+# Copyright 2020-2022 Lowdefy, Inc
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+_ref:
+  path: templates/actions.yaml.njk
+  vars:
+    pageId: DownloadCsv
+    pageTitle: DownloadCsv
+    filePath: actions/DownloadCsv.yaml
+    types: |
+      ```
+      (params: {
+        filename: string,
+        data: any[],
+        fields: string[]
+      }): void
+      ```
+    description: |
+      The `DownloadCsv` action is used to generate and download a CSV file from a set of data.
+    params: |
+      ###### object
+        - `filename: string`: __Required__ - The name for the generated CSV file.
+        - `data: any[]`: __Required__ - The set of data used to generate the CSV file.
+        - `fields: string[]`: __Required__ - The field definitions the set of data will be mapped to in the generated CSV file.
+    examples: |
+      ###### Generate a CSV using an array of data:
+      ```yaml
+      - id: generate_csv
+        type: Button
+        properties:
+          size: large
+          title: Generate a CSV
+          color: '#1890ff'
+        events:
+          onClick:
+            - id: generate_csv
+              type: DownloadCsv
+              params:
+                filename: profiles.csv
+                data:
+                  - Username: booker12 # an array of data, usually a reference
```

Comment:
Think this should be `request` instead of `reference` here and in the following lines